responses/registries.py

```diff
@@ -67,3 +67,25 @@ def replace(self, response: "BaseResponse") -> None:
                 "Response is not registered for URL {}".format(response.url)
             )
         self.registered[index] = response
+
+
+class OrderedRegistry(FirstMatchRegistry):
+    def find(
+        self, request: "PreparedRequest"
+    ) -> Tuple[Optional["BaseResponse"], List[str]]:
+
+        if not self.registered:
+            return None, ["No more registered responses"]
+
+        response = self.registered.pop(0)
+        match_result, reason = response.matches(request)
+        if not match_result:
+            self.reset()
+            self.add(response)
+            reason = (
+                "Next 'Response' in the order doesn't match "
+                f"due to the following reason: {reason}."
```

Comment:
This is a pretty good error message. I think it could be better if we included what the next expected response URL is as well.

Reply:
@markstory 
we get something like in test 
```
"- GET http://twitter.com/api/1/barfoo Next 'Response' in the order doesn't match due to the following reason: URL does not match"
```

do you want to add at the end smth like: `Expected: <url>` ?

Reply:
now this looks like this, further changes in the message will require changes in `__init__.py` which I hesitate a bit.

However, I am fine to change wording in the `reason`

```
Connection refused by Responses - the call doesn't match any registered mock.

Request: 
- GET http://twitter.com/api/1/foobar

Available matches:
- GET http://twitter.com/api/1/barfoo Next 'Response' in the order doesn't match due to the following reason: URL does not match.
```

Reply:
> do you want to add at the end smth like: Expected: <url> ?

I think that would be better. The current error message you have where only one response is listed as available is pretty good. I wonder if developers will be tripped up by some of their mocks missing in the output.

Reply:
@markstory 
the problem is that then we need to rewrite current messaging.
here https://github.com/getsentry/responses/blob/0d2c4f9543fcdf445c8a838ba1b756f007fac02d/responses/__init__.py#L727 we iterate over all registered and report the issue in format (URL - Reason)

if we would like to change it, then it may affect tests that rely on the wording

in this scenario we are flexible only on changing the reason but considering that `Available matches` phrase will be still present.
ideas on meaningful output we can send?

Reply:
generally, it should be somehow logical, that in `OrderedRegistry` we have only one available match since all other are irrelevant

Reply:
I don't have strong conviction that the message will be unclear. How about we move forward with what we have today and wait for feedback that the messaging is unclear or confusing.

Reply:
@markstory 
sure,
maybe somebody will propose a nice alternative :)